raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientReconfigTest.java

```diff
@@ -2130,6 +2136,444 @@ void testFollowerSendsUpdateVoter() throws Exception {
         context.assertFetchRequestData(fetchRequest, epoch, 0L, 0);
     }
 
+    @ParameterizedTest
+    @EnumSource(value = Errors.class, names = {"NONE", "UNSUPPORTED_VERSION"})
+    void testFollowerSendsUpdateVoterWithKraftVersion0(Errors updateVoterError) throws Exception {
+        ReplicaKey local = replicaKey(randomReplicaId(), true);
+        ReplicaKey voter1 = replicaKey(local.id() + 1, true);
```

Comment:
should we have a version of this test where the voters start out w/o directory ids?

Reply:
Yes. I added a test for this.